### ISSUE TITLE
JIRA collector breaks down after some time with logging exception.

### ISSUE THREAD

**Issue by rohitstatic** [capitalone/Hygieia#993]
Hi,

The jira collector works fine for some time and then breaks down with the following exception.

```
Exception in thread "Thread-33" java.lang.NoClassDefFoundError: ch/qos/logback/classic/spi/ThrowableProxy
        at ch.qos.logback.classic.spi.LoggingEvent.<init>(LoggingEvent.java:121)
        at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:440)
        at ch.qos.logback.classic.Logger.filterAndLog_0_Or3Plus(Logger.java:396)
        at ch.qos.logback.classic.Logger.log(Logger.java:788)
        at org.apache.commons.logging.impl.SLF4JLocationAwareLog.error(SLF4JLocationAwareLog.java:216)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient.doExecute(AbstractHttpAsyncClient.java:466)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient.access$000(AbstractHttpAsyncClient.java:101)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient$1.run(AbstractHttpAsyncClient.java:485)
Caused by: java.lang.ClassNotFoundException: ch.qos.logback.classic.spi.ThrowableProxy
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at org.springframework.boot.loader.LaunchedURLClassLoader.doLoadClass(LaunchedURLClassLoader.java:170)
        at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:136)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 8 more
Exception in thread "Thread-3" java.lang.NoClassDefFoundError: org/apache/http/impl/nio/reactor/ExceptionEvent
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.addExceptionEvent(AbstractMultiworkerIOReactor.java:265)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.addExceptionEvent(AbstractMultiworkerIOReactor.java:274)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.execute(AbstractMultiworkerIOReactor.java:396)
        at org.apache.http.impl.nio.conn.PoolingClientAsyncConnectionManager.execute(PoolingClientAsyncConnectionManager.java:108)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient.doExecute(AbstractHttpAsyncClient.java:464)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient.access$000(AbstractHttpAsyncClient.java:101)
        at org.apache.http.impl.nio.client.AbstractHttpAsyncClient$1.run(AbstractHttpAsyncClient.java:485)
Caused by: java.lang.ClassNotFoundException: org.apache.http.impl.nio.reactor.ExceptionEvent
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at org.springframework.boot.loader.LaunchedURLClassLoader.doLoadClass(LaunchedURLClassLoader.java:170)
        at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:136)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

I am running the collector with nohup as follows.
```
nohup java -jar <<jarname>> --spring.config.location=application.properties& > /dev/null
```
Any help here?

**Comment by MarkRx** [comment]
Are you running out of memory perhaps?

https://github.com/spring-projects/spring-boot/issues/4968

**Comment by rohitstatic** [comment]
This got fixed after an upgrade to spring boot 1.3.8 .
This was the root casue. https://github.com/spring-projects/spring-boot/pull/6683
